### ISSUE TITLE
send_file encoding problem (Python3 & OSX)

### ISSUE THREAD

**Issue by johaven** [pallets/werkzeug#1121]
Python: 3.6.1
Werkzeug: 0.12.1

It's happen with files name (with accent and " ' ") on OSX (which use special encoding, i suppose)
I have tried to normalize string with unicodedata (NFC) but nothing to do ...

Content-Disposition: attachment; filename="**Capture d’écran 2017-05-01.png**"
Content-Length: 3498
Content-Type: image/png
Last-Modified: Mon, 08 May 2017 09:17:37 GMT
Cache-Control: public, max-age=43200
Expires: Wed, 17 May 2017 07:09:49 GMT
ETag: "1494235057.0-3498-2239568018"
Date: Tue, 16 May 2017 19:09:49 GMT
Accept-Ranges: bytes
```python
Error on request:
Traceback (most recent call last):
  File "/.virtualenvs/web2/lib/python3.6/site-packages/werkzeug/serving.py", line 209, in run_wsgi
    execute(self.server.app)
  File "/.virtualenvs/web2/lib/python3.6/site-packages/werkzeug/serving.py", line 200, in execute
    write(data)
  File "/virtualenvs/web2/lib/python3.6/site-packages/werkzeug/serving.py", line 168, in write
    self.send_header(key, value)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/http/server.py", line 508, in send_header
    ("%s: %s\r\n" % (keyword, value)).encode('latin-1', 'strict'))
UnicodeEncodeError: 'latin-1' codec can't encode character '\u2019' in position 52: ordinal not in range(256)
```

**Comment by davidism** [comment]
Fixed in Flask with pallets/flask#2223 and pallets/flask#2238.

**Comment by davidism** [comment]
Going to close this as by the time `send_header` is reached the header should already be formatted correctly. `send_file` in Flask does such formatting, but I'm not sure we can generalize it. If someone thinks a `encode_utf8_header_value` function would be useful, they can contribute it.
